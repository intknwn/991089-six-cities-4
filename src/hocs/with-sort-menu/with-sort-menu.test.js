import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import withSortMenu from './with-sort-menu.jsx';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const MockComponent = ({onSortClick}) => {
  return (
    <div>
      <li onClick={() => onSortClick()}></li>
    </div>
  );
};

MockComponent.propTypes = {
  onSortClick: PropTypes.func.isRequired
};

const MockComponentWrapped = withSortMenu(MockComponent);

it(`withSortMenu render`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      sortType: `top rated`,
    },
  });

  const tree = renderer.create((
    <Provider store={store}>
      <MockComponentWrapped
        onSortClick={() => {}}
      />
    </Provider>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});

