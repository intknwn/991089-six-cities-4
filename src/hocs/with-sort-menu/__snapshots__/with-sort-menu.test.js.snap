// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSortMenu render 1`] = `
<div>
  <li
    onClick={[Function]}
  />
</div>
`;
